pkg/odo/cli/component/list.go

```diff
@@ -46,28 +57,53 @@ func NewListOptions() *ListOptions {
 
 // Complete completes log args
 func (lo *ListOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	lo.isExperimentalMode = experimental.IsExperimentalModeEnabled()
+	lo.devfilePath = filepath.Join(lo.componentContext, DevfilePath)
+	if lo.isExperimentalMode && util.CheckPathExists(lo.devfilePath) {
+		// Add a disclaimer that we are in *experimental mode*
+		log.Experimental("Experimental mode is enabled, use at your own risk")
+
+		lo.Context = genericclioptions.NewDevfileContext(cmd)
+		lo.Client = genericclioptions.Client(cmd)
+		lo.hasDCSupport, err = lo.Client.IsDeploymentConfigSupported()
 
-	if util.CheckKubeConfigExist() {
-		klog.V(4).Infof("New Context")
-		lo.Context = genericclioptions.NewContext(cmd)
 	} else {
-		klog.V(4).Infof("New Config Context")
-		lo.Context = genericclioptions.NewConfigContext(cmd)
+		// here we use the config.yaml derived context if its present, else we use information from user's kubeconfig
+		// as odo list should work in a non-component directory too
+
+		if util.CheckKubeConfigExist() {
+			klog.V(4).Infof("New Context")
+			lo.Context = genericclioptions.NewContext(cmd)
+			lo.hasDCSupport, err = lo.Client.IsDeploymentConfigSupported()
 
+		} else {
+			klog.V(4).Infof("New Config Context")
+			lo.Context = genericclioptions.NewConfigContext(cmd)
+			// for disconnected situation we just assume we have DC support
+			lo.hasDCSupport = true
+
+		}
 	}
+
 	return
 
 }
 
 // Validate validates the list parameters
 func (lo *ListOptions) Validate() (err error) {
 
-	var project, app string
-
 	if len(lo.Application) != 0 && lo.allAppsFlag {
 		klog.V(4).Infof("either --app and --all-apps both provided or provided --all-apps in a folder has app, use --all-apps anyway")
 	}
 
+	if lo.isExperimentalMode {
+		if lo.Context.Application == "" && lo.Context.Project == "" {
```

Comment:
I dont think we're using `lo.Context.Project` in experimental mode. You can check `resolveNamespace()` in context.go 🤔